test/src/test-dimension_labels-read-consistency.cc

```diff
@@ -0,0 +1,354 @@
+/**
+ * @file test-dimension_labels.cc
+ *
+ * @section LICENSE
+ *
+ * The MIT License
+ *
+ * @copyright Copyright (c) 2022 TileDB, Inc.
+ *
+ * Permission is hereby granted, free of charge, to any person obtaining a copy
+ * of this software and associated documentation files (the "Software"), to deal
+ * in the Software without restriction, including without limitation the rights
+ * to use, copy, modify, merge, publish, distribute, sublicense, and/or sell
+ * copies of the Software, and to permit persons to whom the Software is
+ * furnished to do so, subject to the following conditions:
+ *
+ * The above copyright notice and this permission notice shall be included in
+ * all copies or substantial portions of the Software.
+ *
+ * THE SOFTWARE IS PROVIDED "AS IS", WITHOUT WARRANTY OF ANY KIND, EXPRESS OR
+ * IMPLIED, INCLUDING BUT NOT LIMITED TO THE WARRANTIES OF MERCHANTABILITY,
+ * FITNESS FOR A PARTICULAR PURPOSE AND NONINFRINGEMENT. IN NO EVENT SHALL THE
+ * AUTHORS OR COPYRIGHT HOLDERS BE LIABLE FOR ANY CLAIM, DAMAGES OR OTHER
+ * LIABILITY, WHETHER IN AN ACTION OF CONTRACT, TORT OR OTHERWISE, ARISING FROM,
+ * OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS IN
+ * THE SOFTWARE.
+ *
+ * @section DESCRIPTION
+ *
+ * Tests dimension labels only read fragments that exist in both labelled and
+ * indexed array.
+ */
+
+#include <algorithm>
+
+#include "test/src/helpers.h"
+#include "test/src/vfs_helpers.h"
+#include "tiledb/api/c_api/context/context_api_internal.h"
+#include "tiledb/sm/array_schema/dimension_label_schema.h"
+#include "tiledb/sm/dimension_label/dimension_label.h"
+#include "tiledb/sm/enums/encryption_type.h"
+#include "tiledb/sm/enums/label_order.h"
+#include "tiledb/sm/enums/query_status.h"
+#include "tiledb/sm/enums/query_type.h"
+#include "tiledb/sm/query/query.h"
+#include "tiledb/sm/query/query_buffer.h"
+#include "tiledb/storage_format/uri/parse_uri.h"
+
+using namespace tiledb::sm;
+using namespace tiledb::test;
+
+/**
+ * Example dimension label with fixed-sized labels.
+ *
+ * Dimension label summary:
+ * * Order: Increasing
+ * * Index dimension: type=uint64_t, domain=[1, 4]
+ * * Label dimesnion: type=uint64_t, domain=[0, 400]
+ *
+ */
+class ExampleFixedDimensionLabel : public TemporaryDirectoryFixture {
```

Comment:
I like what you are doing here a lot, but FWIW two small comments (non-blocking):
- I'm surprised to see the inheritance here -- making a note to discuss w/ you next time we talk because I'd like to better understand how/why this works (haven't used catch2 fixtures very much)
- these read functions look like they should eventually be part of the public C++ API!